pkg/scheduler/core/equivalence/eqivalence.go

```diff
@@ -63,15 +63,57 @@ func NewCache() *Cache {
 type NodeCache struct {
 	mu    sync.RWMutex
 	cache predicateMap
+	// generations stores generation numbers for predicates, incremented on
+	// predicate invalidation. Created on first update. Use 0 if does not
+	// exist.
+	generations map[string]uint64
```

Comment:
I think you can add the snapshot as a member of NodeCache.

Reply:
IIUC, snapshotting generations to current node cache, then in `updateResult` compare previous snapshotted generation with live generation?
It's ok, but requires us to make assumptions how functions are called.

Reply:
@misterikkit @resouer What do you think?